src/serialization/json/JsonParseNode.cs

```diff
@@ -148,17 +150,11 @@ public JsonParseNode(JsonElement node, KiotaJsonSerializationContext jsonSeriali
             if(_jsonNode.ValueKind != JsonValueKind.String)
                 return null;
 
-            if(_jsonNode.TryGetDateTimeOffset(out var dateTimeOffset))
+            if(TryGetUsingTypeInfo(_jsonNode, _jsonSerializerContext.DateTimeOffset, out var dateTimeOffset))
```

Comment:
We should probably implement the same for other date formats. What do you think @andrueastman ?

Reply:
I agree. We probably should. 
The benefit here being that if we use the `_jsonSerializerContext` for parsing first so that customization is possible by passing a custom context. (Currentl not the case for the DateTime and Time types...)

Reply:
Updated for Date and Time